subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/PropertyReportTask.java

```diff
@@ -42,7 +61,17 @@ public void setRenderer(PropertyReportRenderer renderer) {
 
     @Override
     public void generate(Project project) {
-        for (Map.Entry<String, ?> entry : new TreeMap<String, Object>(project.getProperties()).entrySet()) {
+        Map<String, Object> entries = new TreeMap<>(project.getProperties());
+        if (property.isPresent()) {
+            String propertyValue = property.get();
+            if (propertyValue.equals("properties")) {
+                renderer.addProperty(propertyValue, "{...}");
+            } else {
+                renderer.addProperty(propertyValue, entries.get(propertyValue));
+            }
+            return;
```

Comment:
Rather put the `for` block in an `else`, than a sneaky return in the middle? Maybe it's more obvious what's happening?

Reply:
Indeed that's more clean